### ISSUE TITLE
Test Concurrent Download limit of satellites

### ISSUE THREAD

**Issue by littleskunk** [storj/storj#4887]
> How many concurrent downloads can the satellite handle? Now you might wonder isn't that the same as number 1? Not exactly. Number 1 comes with the assumption that a customer is trying to download the same file too many times in parallel. In that situation I expect the storage nodes to be the bottleneck. So let's change the setup just a little bit. Let's say the customer starts the first big file download and adding more concurrent downloads over time. That would distribute the downloads over the entire node population. My expectation is that now the satellite will be the bottleneck. To be more precise I expect the database to get slow. Again the question is how much load can we handle before we would have to say no to a potential customer.
> littleskunk

> Oh you make some of the same points I just made above, :daumen_hoch::hautton-2: . The CRDB limit on ops/second on a single object for reads is again very easy to measure experimentally. Lets write something that benchmarks it and push the system to its limit and quantify. Just don't accidentally leave it running for very long :leichtes_lächeln:
> jacob

We want to find the limit with remote segments. The current use case we are targetting for has more big files (1TB+).

CRDB is great in handling multiple SELECT statements or UPDATES on different rows. The bottleneck will be updating the same row over and over again. For our tests this means we should download the same small file over and over again and run this test for at least 3 hours. Let's make it 6 hours to see the impact of order submission.

**Comment by ihaid** [comment]
300 threads 
  Benchmark 1: uplink cp sj://test/file1.txt f.txt
  Time (mean ± σ):      1.192 s ±  0.095 s    [User: 0.099 s, System: 0.056 s]
  Range (min … max):    1.015 s …  1.473 s    20 runs

  enchmark 1: uplink cp sj://test/file1.txt f.txt
  Time (mean ± σ):      1.157 s ±  0.092 s    [User: 0.088 s, System: 0.050 s]
  Range (min … max):    1.028 s …  1.443 s    20 runs

**Comment by littleskunk** [comment]
Out of scope.
